### PR TITLE
example/centos: ami for aarch64

### PR DESCRIPTION
**Pull request by supakeen**
Based on #210.

### CODE REVIEW

**Comment by schuellerf** [comment]
Not sure if I got it right - should this be working?
`otk compile < example/centos/centos-9-aarch64-ami.yaml`

**Comment by supakeen** [comment]
> Not sure if I got it right - should this be working? `otk compile < example/centos/centos-9-aarch64-ami.yaml`

You need the externals; the easiest way to get those isn't very easy but:

```
# install an editable version of the codebase in a virtual environment
python3 -m venv venv
. venv/bin/pip activate
pip install -e.
```

```
# pull in go externals, also copies the python externals
make external
```

After this all relevant externals will be in `./external` and you can run otk with: `OTK_EXTERNAL_PATH="./external" otk compile path/to/omnifest`

**Comment by schuellerf** [comment]
so as I also mentioned in https://github.com/osbuild/otk/pull/220#issuecomment-2371640318 we'll add it to `.gitignore` right?

**Comment by schuellerf** [comment]
> You need the externals; …

I did that - still I get

<details>
<summary>stacktrace</summary>

```
otk compile < example/centos/centos-9-aarch64-ami.yaml
Traceback (most recent call last):
  File "/…/otk/src/otk/transform.py", line 204, in process_include
    with path.open(encoding="utf8") as fp:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/pathlib.py", line 1013, in open
    return io.open(self, mode, buffering, encoding, errors, newline)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/proc/389292/fd/common/repositories.yaml'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/…/otk/venv/bin/otk", line 8, in <module>
    sys.exit(root())
             ^^^^^^
  File "/…/otk/src/otk/command.py", line 13, in root
    return run(sys.argv[1:])
           ^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/command.py", line 27, in run
    return compile(arguments)
           ^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/command.py", line 79, in compile
    return _process(arguments, dry_run=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/command.py", line 49, in _process
    doc = Omnifest(path)
          ^^^^^^^^^^^^^^
  File "/…/otk/src/otk/document.py", line 31, in __init__
    tree = process_include(self._ctx, state, path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 210, in process_include
    return resolve(ctx, state.copy(path=path), data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 37, in resolve
    return resolve_dict(ctx, state, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 71, in resolve_dict
    process_defines(ctx, state, val)
  File "/…/otk/src/otk/transform.py", line 184, in process_defines
    process_defines(ctx, new_state, value)
  File "/…/otk/src/otk/transform.py", line 184, in process_defines
    process_defines(ctx, new_state, value)
  File "/…/otk/src/otk/transform.py", line 178, in process_defines
    new_vars = resolve(ctx, state, call(key, resolve(ctx, state, value)))
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 37, in resolve
    return resolve_dict(ctx, state, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 120, in resolve_dict
    tree[key] = resolve(ctx, state, val)
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 37, in resolve
    return resolve_dict(ctx, state, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 95, in resolve_dict
    included = process_include(ctx, state, pathlib.Path(val))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/…/otk/src/otk/transform.py", line 207, in process_include
    raise FileNotFoundError(f"file {path} referenced from {state.path} was not found") from fnfe
FileNotFoundError: file /proc/389292/fd/common/repositories.yaml referenced from /proc/self/fd/0 was not found

```
</details>

if it should work, I'll investigate myself

**Comment by supakeen** [comment]
This seems to me that you're using input redirection and include paths are getting based on those which indeed won't work (and is likely a separate bug).

**Review by mvo5 — COMMENTED** [review]
This looks very nice, thank you! It does not include the updated "images" that also add "exclude:" packages yet, right? If so, should we add it here or via a followup?

**Review by mvo5 — DISMISSED** [review]
I guess we should get it in and do the exclude in a followup to move things along.

File: test/data/images-ref/centos/9/x86_64/ami/centos_9-x86_64-ami-empty.json [hunk 1]
```diff
@@ -196,6 +196,96 @@
                 },
                 {
                   "id": "sha256:90aebae315675cbf04612de4f7d5874850f48e0b8dd82becbeaa47ca93f5ebfb"
+                },
```
Comment (mvo5) [inline test/data/images-ref/centos/9/x86_64/ami/centos_9-x86_64-ami-empty.json:199]: Can this maybe be a separate PR or commit? The existing commit says: `ami for aarch64` but this change seems unrelated to the commit description

Comment (supakeen) [reply test/data/images-ref/centos/9/x86_64/ami/centos_9-x86_64-ami-empty.json:199]: Yes, I think this was accidental. Let me verify.

**Review by mvo5 — APPROVED** [review]
Thank you
